contracts/helpers/helpers.ts

```diff
@@ -272,7 +272,7 @@ export async function addStake(
 
         const results = await validatorClient
             .newGroup()
-            .gas({ args: [], staticFee: AlgoAmount.MicroAlgos(0) })
+            .gas({ args: [], staticFee: AlgoAmount.MicroAlgos(0), validityWindow: 200 })
```

Comment:
Are these validity window changes related to the problems you were having with the uncached suggestedParams or is this an unrelated change?

Reply:
Indirectly, yes.  Some of this may be left over as a dart-throw tbh, but on addStake/removeStake/epochBalanceUpdate/addPool - the first/lastValid ranges must NOT be large but must be larger than 11 or so because if in a new 'day', the contracts themselves get the average block time from the chain itself (taking 10 block sample). 
for the block[xx].timestamp opcode, its all based on first/lastValid - not current round - so first/lastValid must be very explicitly controlled in their range.